### DISCUSSION TITLE
Invalid Id Token Nonce

### DISCUSSION THREAD

**Discussion by DBLouis** [zitadel/zitadel#9023] [General]
Hi, having a bit a trouble trying to setup oauth on Jitsi with Zitadel. I am not completely sure if the issue lies on Zitadel side.
I am using https://github.com/MarcelCoding/jitsi-openid to bridge between Zitadel and JWT authentication for Jitsi.
After the redirect I get this error: `Invalid Id Token Nonce: Invalid audiences: 'xxx' is not a trusted audience.`
I created an issue on jitsi-openid which has more details of the setup: https://github.com/MarcelCoding/jitsi-openid/issues/341.

**Comment by fforootd** [comment]
Hm this could be because we insert the projectId and all the clientIds from the same project into the aud claim.

Is there a way in Jitsi to add additional aud claims it will accept?

**Reply by DBLouis** [reply]
I think it is possible but I tried different values and it always fails. I'm using a program to bridge between Oidc and Jitsi Jwt, so could it be that it is not handling this well?

**Reply by fforootd** [reply]
Hm, do you know is that error you see from jitsi or the `jitsi-openid`?

**Reply by DBLouis** [reply]
How can I tell?

**Reply by MarcelCoding** [reply]
It's from jitsi-openid. Just do a code search using GitHub and you see the condition which is triggering the error.

**Reply by DBLouis** [reply]
So jitsi-openid is not happy about the response from Zitadel. If I create a client in a separate project without any other clients, then the list of claims will not contains other client IDs. Maybe that helps.

**Reply by fforootd** [reply]
I quickly glanced at the code and it looks like the rust oidc library is unhappy with us adding multiple ids to the aud claim. To my understanding the rust client can be configured to cope for that but I think this is the same problem we have seen with proxmox recently in our community.

**Reply by fforootd** [reply]
relates to #9200

**Reply by JosiahParry** [reply]
> "...To my understanding the rust client can be configured to cope..."

I think I am running into this issue with openidconnect Rust crate. I am able to use the code flow with Auth0 OIDC but not with Zitadel and am running into "error=Invalid audiences: \`numbers here\` is not a trusted audience"

is the a documented way to address this?

**Reply by anlumo** [reply]
> I think I am running into this issue with openidconnect Rust crate. I am able to use the code flow with Auth0 OIDC but not with Zitadel and am running into "error=Invalid audiences: `numbers here` is not a trusted audience"
> 
> is the a documented way to address this?

I fixed that using

```rs
let id_token_verifier = client.id_token_verifier().set_other_audience_verifier_fn(|_| true);
```

where `client` is a `openidconnect::client::Client`.

I don't know about the security implications of that, though. You can also specifically check for the ids in the closure.

**Reply by JosiahParry** [reply]
@anlumo I made a PR to address this: https://github.com/ramosbugs/openidconnect-rs/pull/212/files
looks like the check in the presence of multiple audiences had some incorrect logic.
